ecogvis/functions/subFunctions.py

```diff
@@ -747,28 +769,36 @@ def BadChannelDel(self, ch_list):
         ch_list : list of integers
             List of indices of channels to be un-marked as 'bad'.
         """
+        # Update list of bad electrodes ids
         for ch in ch_list:
-            if ch in self.badChannels:
-                self.badChannels.remove(ch)
+            if ch in self.bad_channels_ids:
+                self.bad_channels_ids.remove(ch)
+
+        # List of electrodes IDs
+        elecs_ids = list(self.nwb.electrodes.id[:])
+        is_bad_list = [False] * len(elecs_ids)
+        for i, id in enumerate(elecs_ids):
+            if id in self.bad_channels_ids:
+                is_bad_list[i] = True
+
+        if 'bad' not in self.nwb.electrodes:
+            self.nwb.add_electrode_column(
+                name='bad',
+                description='electrode identified as too noisy',
+                data=is_bad_list,
+            )
+        else:
+            self.nwb.electrodes['bad'].data[:] = is_bad_list
+
```

Comment:
This is the same change as above. Try to avoid copy/pasting code. Can this be refactored into a function?

Reply:
fixed with 2f99a2c